Makefile

```
@@ -184,6 +188,7 @@ vcfplugin.o: EXTRA_CPPFLAGS += -DPLUGINPATH='"$(pluginpath)"'
 
 %.dll %.cygdll: %.c version.h version.c libbcftools.a $(HTSLIB_DLL)
 	$(CC) $(PLUGIN_FLAGS) $(CFLAGS) $(ALL_CPPFLAGS) $(EXTRA_CPPFLAGS) $(LDFLAGS) -o $@ version.c $< $(PLUGIN_LIBS)
```

All the `W32_PLUGIN_LIBS` hackery in _Makefile_ and _config.mk.in_ is so that you can get `$(HTSLIB_LDFLAGS)` into this compile&link command. But actually the bug is that it should be there already!

```suggestion
	$(CC) $(PLUGIN_FLAGS) $(CFLAGS) $(ALL_CPPFLAGS) $(EXTRA_CPPFLAGS) $(ALL_LDFLAGS) -o $@ version.c $< $(PLUGIN_LIBS)
```

---

No, the hackery is so we can get the right options in on Windows for builds against the source tree and builds against an installed HTSlib.  They differ slightly and are also different to the ones needed on Linux / Macs, so the solution is to set a variable with appropriate settings in `config.mk` and then apply it for the Windows builds only in `Makefile`.

This solution is probably not optimal, but it does work on all platforms.  I expect we will revisit it later when we have more time.

---

Oops, didn't notice it was `$(HTSLIB_DLL)` on one side and `$(HTSLIB_LIB)` on the other. Fair enough until the glorious day when it is revisited.